core/src/main/java/cz/o2/proxima/repository/RepositoryFactory.java

```diff
@@ -28,8 +31,12 @@
 
     private final RepositoryFactory underlying;
 
-    private Caching(RepositoryFactory underlying) {
+    private Caching(RepositoryFactory underlying, Repository created) {
       this.underlying = underlying;
+      synchronized (Repository.class) {
```

Comment:
Looking at this class, I don't really think that any of the fields should be in static context. What if you have multiple repositories per JVM? This doesn't look right..